sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala

```diff
@@ -189,7 +188,7 @@ trait ProgressReporter extends Logging {
       sink = sinkProgress,
       observedMetrics = new java.util.HashMap(observedMetrics.asJava))
 
-    if (hasNewData) {
+    if (hasExecuted) {
```

Comment:
This was also incorrect for no new data micro batches

Reply:
Nice finding. We don't recognize the bug because lastNoDataProgressEventTime is set to Long.MinValue which makes next no new data micro batch to update the progress immediately, which hides the bug. (If that's intentional, well, then it's too tricky and we should have commented here.)

Maybe we should also rename lastNoDataProgressEventTime as well as the fix changes the semantic?

And we may want to revisit that our intention is updating progress immediately whenever the batch has not run after any batch run.

Reply:
oh, that's why I'm facing issues... I understand better now